audit/conn/domainsocket.go

```diff
@@ -0,0 +1,77 @@
+//go:build linux || darwin
+
+package conn
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/microsoft/go-otel-audit/audit/conn/internal/writer"
+)
+
+// Compile check on interface implementation.
+var _ Audit = DomainSocketConn{}
+
+// DomainSocketConn represents a connection to a remote audit server via a unix domain socket
+// set to SOCK_STREAM. This implements conn.Audit interface.
+type DomainSocketConn struct {
+	*writer.Conn
+}
+
+// Type returns the type of the audit connection.
+func (DomainSocketConn) Type() Type {
+	return TypeDomainSocket
+}
+
+func (DomainSocketConn) private() {}
+
+type dsOptions struct {
+	path string
+}
+
+func (o *dsOptions) setDefaults() {
+	const (
+		asaSocket    = "/var/run/mdsd/asa/default_fluent.socket"
+		fluentSocket = "/var/run/mdsd/default_fluent.socket"
+	)
+	if o.path != "" {
+		return
+	}
+
+	o.path = asaSocket
+	_, err := os.Stat(o.path)
+	if err != nil {
+		o.path = fluentSocket
+	}
+}
+
+// DSOption is an option for NewDomainSocket.
+type DSOption func(*dsOptions) error
+
+// DSPath sets the path to the unix domain socket. This is usually automatically detected and should
+// only be overridden in very specific circumstances.
+func DSPath(path string) DSOption {
+	return func(o *dsOptions) error {
+		o.path = path
+		return nil
+	}
+}
+
+// NewDomainSocket creates a new connection to the remote audit server.
+func NewDomainSocket(options ...DSOption) (DomainSocketConn, error) {
```

Comment:
Why is this public but the tcp conn internal?

Reply:
Either I don't understand the question or one of us is confused.

Output of tree:
```bash
.
├── conn.go
├── conn_test.go
├── domainsocket.go
├── internal
│   ├── server
│   │   └── server.go
│   └── writer
│       ├── writer.go
│       └── writer_test.go
├── noop.go
├── tcp.go
└── type_string.go
```
Both should be outside of internal.

Reply:
I just cant read at the end of the day, lol